docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,101 @@
+Some time ago, we have added an experimental API in this package that will allow us using CSS variables in the styles of the components.
+Now, we want to try and utilize this new feature in one of the Material UI components.
+
+> If you are curious to learn the benefits you would get by using CSS variables, check out the [RFC](https://github.com/mui/material-ui/issues/27651) for CSS variables support.
+
+## Usage
+
+At this moment, the API does not require any breaking changes for the Material UI components, however it depends on using a new experimental provider for the theme, called `Experimental_CssVarsProvider`.
+Except for providing the theme in the inner React context, this new provider has as a responsibility to generate CSS variables out of all tokens in the theme that are not functions, and make them available in the context.
+All these variables, are available under a key in the theme, named `vars`.
+The structure of this object is identical to the theme structure, the only difference is that the values represent some CSS varaibles.
+
+The best way to see this is by example.
+Let's say we have this element that uses some theme token:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivThemeToken.js", "hideToolbar": true }}
+
+If you try to inspect this element, you will see that the color style have directly a hex value, that comes some theme token:
+
+```html
+<div style="color: #2e7d32;">...</div>
+```
+
+On the other hand, if the div uses some CSS variable as in this example:The one using the CSS variables however, allows developers to track the path of where the token is defined in theme:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.vars.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivCSSVariable.js", "hideToolbar": true }}
+
+```html
+<div style="color: var(--md-palette-success-main);">...</div>
+```
+
+More over, if you inspect the styles, you can click on the CSS variable and see the actual color value.
+We believe this will significantly increase the DX when debugging the styles for the component.
+
+### Custom theme
+
+The example above uses the default theme. In the next example you will learn how to create your custom theme, using the `experimental_extendTheme` utility.
+
+Before starting with the example, it's important to distingwish some differences in the theme structure.
+The new theme structure, lets you define all different themes you have in one theme, by defining different color schemes.
+Then, when you want to change the app to use different theme, you can use the `useColorScheme` to set the color scheme you want to use.
+This will cause the browser to re-write the CSS varibles to point to the new values, while the components won't change at all, as they should still link to the same variables).
```

Comment:
```suggestion
The example above is using the default Material UI theme. But, what if you wanted to create your own custom theme? Let's see how to do that using the `experimental_extendTheme` utility. 

To start, it is important to understand this new theme structure. A theme is a collection of color schemes, all defined within a single theme object. You'll be able to create more than light and dark color schemes if you want but for the sake of simplicity, let's use that as an example. Here's how you'd customize it:

```jsx
const theme = experimental_extendTheme({
  colorSchemes: {
    light: {
      palette: {
        primary: teal,
        secondary: deepOrange,
      },
    },
    dark: {
      palette: {
        primary: cyan,
        secondary: orange,
      },
    },
  },
});
```

If you want to be able to toggle between color schemes, you can use the `useColorScheme` hook. This will make the browser re-write the CSS variables, pointing them to the new values you've created. The components won't change at all, though—they should still refer to the same variables as before.
```

Enriching this section a bit more, it's very important! Though I'm still a bit confused with what the last sentence means: _"The components won't change at all, though—they should still refer to the same variables as before."._

Reply:
> Though I'm still a bit confused with what the last sentence means: _"The components won't change at all, though—they should still refer to the same variables as before."._

The component's styles are not changing at all (the same classname is applied, as none of the styles changed), only the values for the CSS variable changed.

What I mean with this is, previously it worked like this:

1. light mode:
.some-class {
  color: #123456
}
2. dark mode:
.some-other-class {
  color: #654321
}

Not it works like this:

1. light mode:
.some-class {
  color: var(--md-palette-primary-main); // #123456
}
2. dark mode:
.some-class {
  color: var(--md-palette-primary-main); // #654321
}

Reply:
Alright, just pushed a refinement on this part! Thanks for explaining :)